conda-store-server/conda_store_server/server/views/api.py

```diff
@@ -688,6 +688,60 @@ async def api_put_build(
     }
 
 
+@router_api.put(
+    "/build/{build_id}/cancel/",
+    response_model=schema.APIAckResponse,
+)
+async def api_put_build_cancel(
+    build_id: int,
+    request: Request,
+    conda_store=Depends(dependencies.get_conda_store),
+    db: Session = Depends(dependencies.get_db),
+    auth=Depends(dependencies.get_auth),
+):
+    build = api.get_build(db, build_id)
+    if build is None:
+        raise HTTPException(status_code=404, detail="build id does not exist")
+
+    auth.authorize_request(
+        request,
+        f"{build.environment.namespace.name}/{build.environment.name}",
+        {Permissions.BUILD_CANCEL},
+        require=True,
+    )
+
+    if conda_store.celery_app.control.inspect().ping() is None:
+        raise HTTPException(
+            status_code=409,
+            detail="conda-store celery broker does not support task cancelation. Use redis or rabbitmq message queues. See docs for a more detailed explanation",
+        )
+
+    conda_store.celery_app.control.revoke(
+        [
+            f"build-{build_id}-conda-env-export",
+            f"build-{build_id}-conda-pack",
+            f"build-{build_id}-conda-docker",
+            f"build-{build_id}-environment",
+        ],
```

Comment:
I see what's done here. We're cancelling all the possibly existing tasks for the given build_id. 

There's a problem with `f"build-{build_id}-conda-docker"` : The name of the task is actually `f"build-{build.id}-docker"`, based on [app.py](https://github.com/conda-incubator/conda-store/blob/67364dcd65f2b50fbea3be5be7dc4fbaadf2b7ab/conda-store-server/conda_store_server/app.py#L674)

Overall, I think we could provide a more resilient approach, where we don't have to manually maintain a list of tasks here in `api.py`, to match with `app.py`

Short term solution : fix the wrong task name above
Longer term solution : Use constants or something so the list above is generated automatically and identical to the strings used when creating the tasks, removing the need to maintain it.

Reply:
Totally. This makes sense. 

I'm wondering where the best place to store this is. Currently the most natural place feels like `orm.py`. 

Also related to this I am not sure if all those task names needed to be included? This is something that confuses me in celery. The "root" task name is `build-{build_id}-environment`

Reply:
@pierrotsmnrd I don't think that celery considers the children subtasks to "exist" and be cancelable/revokable. This is what needs to be checked.

Reply:
@anirrudh can you check this?

Reply:
Yeah, I will check this in our case. I'll have some thoughts in a bit.

Reply:
@costrouc I've experimented a bit, and this seems to be the case. Celery does not list these as part of tasks that are collected in the overall task id. I tried, and we would have to use something like `celery.chord` at task orchestration time, and then pull this information out later from the object.

Reply:
- I agree that task ids shouldn't be hardcoded. I'll make this change.
- I'll see if it's possible to cancel subtasks using `build-{build.id}` and/or whether this task id should be canceled as well.